lms/djangoapps/course_blocks/transformers/user_partitions.py

```diff
@@ -81,45 +81,46 @@ def collect(cls, block_structure):
     def transform_block_filters(self, usage_info, block_structure):
         user = usage_info.user
         result_list = SplitTestTransformer().transform_block_filters(usage_info, block_structure)
+        staff_access = has_access(user, 'staff', usage_info.course_key)
 
         user_partitions = block_structure.get_transformer_data(self, 'user_partitions')
         if not user_partitions:
             return [block_structure.create_universal_filter()]
 
         user_groups = get_user_partition_groups(usage_info.course_key, user_partitions, user, 'id')
 
-        for block_key in block_structure.topological_traversal():
-            transformer_block_field = block_structure.get_transformer_block_field(
-                block_key, self, 'merged_group_access'
-            )
-            access_denying_partition_id = transformer_block_field.get_access_denying_partition(
-                user_groups
-            )
-            access_denying_partition = get_partition_from_id(user_partitions, access_denying_partition_id)
-
-            if not has_access(user, 'staff', block_key) and access_denying_partition:
-                user_group = user_groups.get(access_denying_partition.id)
-                allowed_groups = transformer_block_field.get_allowed_groups()[access_denying_partition.id]
-                access_denied_message = access_denying_partition.access_denied_message(
-                    block_key, user, user_group, allowed_groups
+        if not staff_access:
```

Comment:
Mike Terry would say that this should just include an early return for `if staff_access:` so that way you don't have to leave this whole path indented. I agree with him

Reply:
Prefer the following?

```
if staff_access:
  return result_list
restofcodehere
return result_list
```

Also not this debate 😭 . I definitely lean toward singular exit points if possible, it makes debugging much simpler, and I'd also argue that it's not any more/less readable being indented or not.

Though maybe it's worthwhile to just add the extra return since there's already a second in this method a third isn't _that_ much worse.

Reply:
Yeah I'll do it 👍